WikipediaUnitTests/MWKSectionHasTextDataTests.m

```diff
@@ -0,0 +1,67 @@
+//  Created by Monte Hurd on 9/22/15.
+//  Copyright (c) 2015 Wikimedia Foundation. All rights reserved.
+
+#import "XCTestCase+MWKFixtures.h"
+#import <XCTest/XCTest.h>
+#import "MWKDataStore+TemporaryDataStore.h"
+#import "WMFTestFixtureUtilities.h"
+#import "SessionSingleton.h"
+
+#define HC_SHORTHAND 1
+#define MOCKITO_SHORTHAND 1
+
+#import <OCHamcrest/OCHamcrest.h>
+#import <OCMockito/OCMockito.h>
+
+@interface MWKSectionHasTextDataTests : XCTestCase
+@property SessionSingleton* session;
+@end
+
+@implementation MWKSectionHasTextDataTests
+
+- (void)setUp {
+    [super setUp];
+    self.session = [[SessionSingleton alloc] initWithDataStore:[MWKDataStore temporaryDataStore]];
+}
+
+- (void)tearDown {
+    [self.session.dataStore removeFolderAtBasePath];
+    [super tearDown];
+}
+
+- (MWKArticle*)getTestingArticle {
+    MWKTitle* title     = [MWKTitle titleWithString:@"Barack Obama" site:[MWKSite siteWithDomain:@"wikipedia.org" language:@"en"]];
+    MWKArticle* article = [self articleWithMobileViewJSONFixture:@"Obama" withTitle:title dataStore:self.session.dataStore];
+    [article save];
+    return article;
+}
+
+- (void)testArticleHasAtLeastOneSectionWithZeroLengthHTML {
+    // Ensure at least one section of article has zero length section html.
+    MWKArticle* article              = [self getTestingArticle];
+    BOOL atLeastOneZeroLengthSection = NO;
+    for (MWKSection* section in article.sections) {
+        if (section.text.length == 0) {
+            atLeastOneZeroLengthSection = YES;
+            break;
+        }
+    }
+    assertThat(@(atLeastOneZeroLengthSection), isTrue());
+}
+
+- (void)testHasTextDataMethodReturnsYESforZeroLengthSectionHTML {
+    // Ensure "[MWKSection hasTextData]" returns YES if section html isn't nil - even if it's a zero length string.
+    MWKArticle* article = [self getTestingArticle];
+    for (MWKSection* section in article.sections) {
+        /*
+           Reminder: zero length strings are *valid* section text data!
```

Comment:
Technically, this might belong in the documentation of `-[MWKSection text]`.  That is, `text` is a non-null string, which implies that an empty string is valid.  The caveat here is that what we're really considering (and there's no great place to document this yet) is that we consider any file written to the data store as a signal that we've downloaded **all** the data for that section.  I hope to eventually write up some high-level architecture documentation about all of this, but the data layer's documentation might have to wait until it's rewritten ;-).